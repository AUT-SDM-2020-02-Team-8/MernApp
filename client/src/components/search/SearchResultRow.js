import React, { Component } from "react"
import { TableRow, TableCell } from "@material-ui/core"

class SearchResultRow extends Component {
  renderCells() {
    let cells = []
    for (const i of this.props.allColumns) {
      if (this.props.chosenColumns.indexOf(i) >= 0) {
        const txt = Array.isArray(this.props.evidence[i]) ? this.props.evidence[i].join(', ') : this.props.evidence[i]
        cells.push(<TableCell key={i}>{txt}</TableCell>)
      }
    }
    return cells
  }

  render() {
    return (
      <TableRow>{this.renderCells()}</TableRow>
    )
  }
}

export default SearchResultRow