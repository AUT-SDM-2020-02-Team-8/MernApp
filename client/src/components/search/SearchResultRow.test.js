import React from 'react';
import { shallow } from 'enzyme';
import SearchResultRow from './SearchResultRow';

test('renders homepage', () => {
  const evidence = {
    title: 'title 1',
    author: 'author 1',
    year: '2020',
    recordType: 'reference paper',
    journal: 'IEEE',
    publisher: 'IEEE',
    sePractice: 'TDD',
    claims: ["code quality", "team confidence", "product quality"]
  }
  const wrapper = shallow(<SearchResultRow evidence={evidence} />);
  expect(wrapper).toMatchSnapshot();
});
