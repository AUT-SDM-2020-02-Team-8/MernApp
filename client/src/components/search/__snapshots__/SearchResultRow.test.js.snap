// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage 1`] = `
<WithStyles(ForwardRef(TableRow))>
  <WithStyles(ForwardRef(TableCell))>
    title 1
  </WithStyles(ForwardRef(TableCell))>
  <WithStyles(ForwardRef(TableCell))>
    author 1
  </WithStyles(ForwardRef(TableCell))>
  <WithStyles(ForwardRef(TableCell))>
    2020
  </WithStyles(ForwardRef(TableCell))>
  <WithStyles(ForwardRef(TableCell))>
    reference paper
  </WithStyles(ForwardRef(TableCell))>
  <WithStyles(ForwardRef(TableCell))>
    IEEE
  </WithStyles(ForwardRef(TableCell))>
  <WithStyles(ForwardRef(TableCell))>
    IEEE
  </WithStyles(ForwardRef(TableCell))>
  <WithStyles(ForwardRef(TableCell))>
    TDD
  </WithStyles(ForwardRef(TableCell))>
  <WithStyles(ForwardRef(TableCell))>
    code quality, team confidence, product quality
  </WithStyles(ForwardRef(TableCell))>
</WithStyles(ForwardRef(TableRow))>
`;
